sdk/storage/azure-storage-file-datalake/tests/test_directory.py

```diff
@@ -95,6 +96,175 @@ def test_create_directory(self, datalake_storage_account_name, datalake_storage_
         # Assert
         self.assertTrue(created)
 
+    @DataLakePreparer()
+    def test_create_file_owner_with_directory_client(self, datalake_storage_account_name, datalake_storage_account_key):
+        self._setUp(datalake_storage_account_name, datalake_storage_account_key)
+        test_string = str(uuid.uuid4())
+        # Arrange
+        directory_name = self._get_directory_reference()
+
+        # Create a directory to put the file under that
+        directory_client = self.dsc.get_directory_client(self.file_system_name, directory_name)
+        directory_client.create_directory()
```

Comment:
I think this file, being `test_directory` should really test the `create_directory` functions. I don't think we have to test `create_file` from directory client since that will be covered in `test_file`. This applies to a few tests in here, let's just have this file testing directories.